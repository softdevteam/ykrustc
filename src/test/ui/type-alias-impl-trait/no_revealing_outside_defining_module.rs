// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete

fn main() {}

mod boo {
    pub type Boo = impl ::std::fmt::Debug;
    fn bomp() -> Boo {
        ""
    }
}

// We don't actually know the type here.

fn bomp2() {
    let _: &str = bomp(); //~ ERROR mismatched types
}

fn bomp() -> boo::Boo {
    "" //~ ERROR mismatched types
}

fn bomp_loop() -> boo::Boo {
    loop {}
}
