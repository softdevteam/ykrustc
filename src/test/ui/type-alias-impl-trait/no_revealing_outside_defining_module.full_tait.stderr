warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/no_revealing_outside_defining_module.rs:3:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error[E0308]: mismatched types
  --> $DIR/no_revealing_outside_defining_module.rs:18:19
   |
LL |     pub type Boo = impl ::std::fmt::Debug;
   |                    ---------------------- the found opaque type
...
LL |     let _: &str = bomp();
   |            ----   ^^^^^^ expected `&str`, found opaque type
   |            |
   |            expected due to this
   |
   = note: expected reference `&str`
            found opaque type `impl Debug`

error[E0308]: mismatched types
  --> $DIR/no_revealing_outside_defining_module.rs:22:5
   |
LL |     pub type Boo = impl ::std::fmt::Debug;
   |                    ---------------------- the expected opaque type
...
LL | fn bomp() -> boo::Boo {
   |              -------- expected `impl Debug` because of return type
LL |     ""
   |     ^^ expected opaque type, found `&str`
   |
   = note: expected opaque type `impl Debug`
                found reference `&'static str`

error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0308`.
